File: mRemoteV1/Config/Import/RemoteDesktopConnection.cs
using System.Collections.Generic;
using System;
using AxWFICALib;
using System.Drawing;
using System.Diagnostics;
using System.Data;
using AxMSTSCLib;
using Microsoft.VisualBasic;
using System.Collections;
using System.Windows.Forms;
using System.IO;
using mRemoteNG.App;


namespace mRemoteNG.Config.Import
{
	public class RemoteDesktopConnection
	{
		public static void Import(string fileName, TreeNode parentTreeNode)
		{
			string[] lines = File.ReadAllLines(fileName);
				
			string name = Path.GetFileNameWithoutExtension(fileName);
			TreeNode treeNode = new TreeNode(name);
			parentTreeNode.Nodes.Add(treeNode);
				
			Connection.ConnectionRecordImp connectionInfo = new Connection.ConnectionRecordImp();
			connectionInfo.Inherit = new Connection.ConnectionRecordImp.ConnectionRecordInheritanceImp(connectionInfo);
			connectionInfo.Name = name;
			connectionInfo.TreeNode = treeNode;
				
			if (treeNode.Parent.Tag is Container.Info)
			{
                connectionInfo.Parent = (Container.Info)treeNode.Parent.Tag;
			}
				
			treeNode.Name = name;
			treeNode.Tag = connectionInfo;
			treeNode.ImageIndex = (int)Images.Enums.TreeImage.ConnectionClosed;
			treeNode.SelectedImageIndex = (int)Images.Enums.TreeImage.ConnectionClosed;
				
			foreach (string line in lines)
			{
				string[] parts = line.Split(new char[] {':'}, 3);
				if (parts.Length < 3)
				{
					continue;
				}
					
				string key = parts[0];
				string value = parts[2];
					
				SetConnectionInfoParameter(connectionInfo, key, value);
			}
				
			Runtime.ConnectionList.Add(connectionInfo);
		}
			
		private static void SetConnectionInfoParameter(Connection.ConnectionRecordImp connectionInfo, string key, string value)
		{
			switch (key.ToLower())
			{
				case "full address":
					Uri uri = new Uri("dummyscheme" + System.Uri.SchemeDelimiter + value);
					if (!string.IsNullOrEmpty(uri.Host))
					{
						connectionInfo.Hostname = uri.Host;
					}
					if (!(uri.Port == -1))
					{
						connectionInfo.Port = uri.Port;
					}
					break;
				case "server port":
					connectionInfo.Port = System.Convert.ToInt32(value);
					break;
				case "username":
					connectionInfo.Username = value;
					break;
				case "domain":
					connectionInfo.Domain = value;
					break;
				case "session bpp":
					switch (value)
					{
						case "8":
							connectionInfo.Colors = Connection.Protocol.RDP.RDPColors.Colors256;
							break;
						case "15":
							connectionInfo.Colors = Connection.Protocol.RDP.RDPColors.Colors15Bit;
							break;
						case "16":
							connectionInfo.Colors = Connection.Protocol.RDP.RDPColors.Colors16Bit;
							break;
						case "24":
							connectionInfo.Colors = Connection.Protocol.RDP.RDPColors.Colors24Bit;
							break;
						case "32":
							connectionInfo.Colors = Connection.Protocol.RDP.RDPColors.Colors32Bit;
							break;
					}
					break;
				case "bitmapcachepersistenable":
					if (value == "1")
					{
						connectionInfo.CacheBitmaps = true;
					}
					else
					{
						connectionInfo.CacheBitmaps = false;
					}
					break;
				case "screen mode id":
					if (value == "2")
					{
						connectionInfo.Resolution = Connection.Protocol.RDP.RDPResolutions.Fullscreen;
					}
					else
					{
						connectionInfo.Resolution = Connection.Protocol.RDP.RDPResolutions.FitToWindow;
					}
					break;
				case "connect to console":
					if (value == "1")
					{
						connectionInfo.UseConsoleSession = true;
					}
					break;
				case "disable wallpaper":
					if (value == "1")
					{
						connectionInfo.DisplayWallpaper = true;
					}
					else
					{
						connectionInfo.DisplayWallpaper = false;
					}
					break;
				case "disable themes":
					if (value == "1")
					{
						connectionInfo.DisplayThemes = true;
					}
					else
					{
						connectionInfo.DisplayThemes = false;
					}
					break;
				case "allow font smoothing":
					if (value == "1")
					{
						connectionInfo.EnableFontSmoothing = true;
					}
					else
					{
						connectionInfo.EnableFontSmoothing = false;
					}
					break;
				case "allow desktop composition":
					if (value == "1")
					{
						connectionInfo.EnableDesktopComposition = true;
					}
					else
					{
						connectionInfo.EnableDesktopComposition = false;
					}
					break;
				case "redirectsmartcards":
					if (value == "1")
					{
						connectionInfo.RedirectSmartCards = true;
					}
					else
					{
						connectionInfo.RedirectSmartCards = false;
					}
					break;
				case "redirectdrives":
					if (value == "1")
					{
						connectionInfo.RedirectDiskDrives = true;
					}
					else
					{
						connectionInfo.RedirectDiskDrives = false;
					}
					break;
				case "redirectcomports":
					if (value == "1")
					{
						connectionInfo.RedirectPorts = true;
					}
					else
					{
						connectionInfo.RedirectPorts = false;
					}
					break;
				case "redirectprinters":
					if (value == "1")
					{
						connectionInfo.RedirectPrinters = true;
					}
					else
					{
						connectionInfo.RedirectPrinters = false;
					}
					break;
				case "audiomode":
					switch (value)
					{
						case "0":
							connectionInfo.RedirectSound = Connection.Protocol.RDP.RDPSounds.BringToThisComputer;
							break;
						case "1":
							connectionInfo.RedirectSound = Connection.Protocol.RDP.RDPSounds.LeaveAtRemoteComputer;
							break;
						case "2":
							connectionInfo.RedirectSound = Connection.Protocol.RDP.RDPSounds.DoNotPlay;
							break;
					}
					break;
			}
		}
	}
}

File: mRemoteV1/Config/Import/RemoteDesktopConnection.vb
Imports System.IO
Imports mRemote3G.App
Imports mRemote3G.Connection
Imports mRemote3G.Connection.Protocol
Imports mRemote3G.Images

Namespace Config.Import
    Public Class RemoteDesktopConnection
        Public Shared Sub Import(fileName As String, parentTreeNode As TreeNode)
            Dim lines As String() = File.ReadAllLines(fileName)

            Dim name As String = Path.GetFileNameWithoutExtension(fileName)
            Dim treeNode = New TreeNode(name)
            parentTreeNode.Nodes.Add(treeNode)

            Dim connectionInfo As New Info
            connectionInfo.Inherit = New Info.Inheritance(connectionInfo)
            connectionInfo.Name = name
            connectionInfo.TreeNode = treeNode

            If TypeOf treeNode.Parent.Tag Is Container.Info Then
                connectionInfo.Parent = treeNode.Parent.Tag
            End If

            treeNode.Name = name
            treeNode.Tag = connectionInfo
            treeNode.ImageIndex = Enums.TreeImage.ConnectionClosed
            treeNode.SelectedImageIndex = Enums.TreeImage.ConnectionClosed

            For Each line As String In lines
                Dim parts() As String = line.Split(New Char() {":"}, 3)
                If parts.Length < 3 Then Continue For

                Dim key As String = parts(0)
                Dim value As String = parts(2)

                SetConnectionInfoParameter(connectionInfo, key, value)
            Next

            Runtime.ConnectionList.Add(connectionInfo)
        End Sub

        Private Shared Sub SetConnectionInfoParameter(ByRef connectionInfo As Info, key As String, value As String)
            Select Case LCase(key)
                Case "full address"
                    Dim uri As New Uri("dummyscheme" + uri.SchemeDelimiter + value)
                    If Not String.IsNullOrEmpty(uri.Host) Then connectionInfo.Hostname = uri.Host
                    If Not uri.Port = - 1 Then connectionInfo.Port = uri.Port
                Case "server port"
                    connectionInfo.Port = value
                Case "username"
                    connectionInfo.Username = value
                Case "domain"
                    connectionInfo.Domain = value
                Case "session bpp"
                    Select Case value
                        Case 8
                            connectionInfo.Colors = RDP.RDPColors.Colors256
                        Case 15
                            connectionInfo.Colors = RDP.RDPColors.Colors15Bit
                        Case 16
                            connectionInfo.Colors = RDP.RDPColors.Colors16Bit
                        Case 24
                            connectionInfo.Colors = RDP.RDPColors.Colors24Bit
                        Case 32
                            connectionInfo.Colors = RDP.RDPColors.Colors32Bit
                    End Select
                Case "bitmapcachepersistenable"
                    If value = 1 Then
                        connectionInfo.CacheBitmaps = True
                    Else
                        connectionInfo.CacheBitmaps = False
                    End If
                Case "screen mode id"
                    If value = 2 Then
                        connectionInfo.Resolution = RDP.RDPResolutions.Fullscreen
                    Else
                        connectionInfo.Resolution = RDP.RDPResolutions.FitToWindow
                    End If
                Case "connect to console"
                    If value = 1 Then
                        connectionInfo.UseConsoleSession = True
                    End If
                Case "disable wallpaper"
                    If value = 1 Then
                        connectionInfo.DisplayWallpaper = True
                    Else
                        connectionInfo.DisplayWallpaper = False
                    End If
                Case "disable themes"
                    If value = 1 Then
                        connectionInfo.DisplayThemes = True
                    Else
                        connectionInfo.DisplayThemes = False
                    End If
                Case "allow font smoothing"
                    If value = 1 Then
                        connectionInfo.EnableFontSmoothing = True
                    Else
                        connectionInfo.EnableFontSmoothing = False
                    End If
                Case "allow desktop composition"
                    If value = 1 Then
                        connectionInfo.EnableDesktopComposition = True
                    Else
                        connectionInfo.EnableDesktopComposition = False
                    End If
                Case "redirectsmartcards"
                    If value = 1 Then
                        connectionInfo.RedirectSmartCards = True
                    Else
                        connectionInfo.RedirectSmartCards = False
                    End If
                Case "redirectdrives"
                    If value = 1 Then
                        connectionInfo.RedirectDiskDrives = True
                    Else
                        connectionInfo.RedirectDiskDrives = False
                    End If
                Case "redirectcomports"
                    If value = 1 Then
                        connectionInfo.RedirectPorts = True
                    Else
                        connectionInfo.RedirectPorts = False
                    End If
                Case "redirectprinters"
                    If value = 1 Then
                        connectionInfo.RedirectPrinters = True
                    Else
                        connectionInfo.RedirectPrinters = False
                    End If
                Case "audiomode"
                    Select Case value
                        Case 0
                            connectionInfo.RedirectSound = RDP.RDPSounds.BringToThisComputer
                        Case 1
                            connectionInfo.RedirectSound = RDP.RDPSounds.LeaveAtRemoteComputer
                        Case 2
                            connectionInfo.RedirectSound = RDP.RDPSounds.DoNotPlay
                    End Select
            End Select
        End Sub

        Private Sub New()
        End Sub
    End Class
End Namespace